### PR TITLE
fix(e2e): stabilize the gatewayapi E2E conformance tests by ignoring headless service cases

### PR DESCRIPTION
**Pull request by jijiechen**
Gateway API conformance tests flakinesses are widely seen, here are two examples:
* https://github.com/kumahq/kuma/actions/runs/10790251438/job/29932838255#step:12:8008
* https://github.com/kumahq/kuma/actions/runs/10933846592/job/30354678824#step:12:6631

This PR tries to stabilize the test suite by ignoring headless service cases temporarily.
It's found by some recent work that these headless service cases are cause differences between successful runs and fail runs. 

With the changes in this PR, we got 4 consecutive successful runs:
https://github.com/kumahq/kuma/actions/runs/10936141162

### Checklist prior to review

- [x] [Link to relevant issue][1] as well as docs and UI issues
  - CI flakiness
- [x] This will not break child repos: it doesn't hardcode values (.e.g "kumahq" as a image registry) and it will work on Windows, system specific functions like `syscall.Mkfifo` have equivalent implementation on the other OS 
  - No
- [x] Tests (Unit test, E2E tests, manual test on universal and k8s) 
  - Tested
  - Don't forget `ci/` labels to run additional/fewer tests
- [x] Do you need to update [`UPGRADE.md`](../blob/master/UPGRADE.md)? --
- [x] Does it need to be backported according to the [backporting policy](../blob/master/CONTRIBUTING.md#backporting)? ([this](https://github.com/kumahq/kuma/actions/workflows/auto-backport.yaml) GH action will add "backport" label based on these [file globs](https://github.com/kumahq/kuma/blob/master/.github/workflows/auto-backport.yaml#L6), if you want to prevent it from adding the "backport" label use [no-backport-autolabel](https://github.com/kumahq/kuma/blob/master/.github/workflows/auto-backport.yaml#L8) label) 
   - No

[1]: https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword

### CODE REVIEW

**Comment by jakubdyszkiewicz** [comment]
if we merge it, let's create an issue to restore it

**Comment by jijiechen** [comment]
The work is made based on the logs collected in these runs:
https://github.com/kumahq/kuma/actions/runs/10920125399/attempts/4

The problematic workload is `echo-v1-dcf569668-*`
* In a successful run, the DP will eventually get cluster `echo-v2_gateway-conformance-mesh_svc_80-9948418762b34993` from the CP
* In a fail run, the DP does not request cluster `echo-v2_gateway-conformance-mesh_svc_80-9948418762b34993` so it does not have it, and it responds `503` to test web requests

**Comment by jijiechen** [comment]
There is an error happens on CP in both successful runs and fail runs.

```log
2024-09-18T11:43:28.188Z	ERROR	vips-outbound-view	there are two external services with the same 'networking.address' or two headless services, to disable automatic DNS generation in external services set 'networking.disableHostDNSEntry=true' in at least one of these externalService	{"error": "autogenerated DNS entry 10.42.0.10:8080 from kubernetes conflicts with existing entry kubernetes"}
github.com/kumahq/kuma/pkg/plugins/runtime/k8s/controllers.KubeHosts
	github.com/kumahq/kuma/pkg/plugins/runtime/k8s/controllers/kube_hosts_converter.go:55
github.com/kumahq/kuma/pkg/plugins/runtime/k8s/controllers.(*ConfigMapReconciler).Reconcile
	github.com/kumahq/kuma/pkg/plugins/runtime/k8s/controllers/configmap_controller.go:55
sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller[...]).Reconcile
	sigs.k8s.io/controller-runtime@v0.19.0/pkg/internal/controller/controller.go:116
sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller[...]).reconcileHandler
	sigs.k8s.io/controller-runtime@v0.19.0/pkg/internal/controller/controller.go:303
sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller[...]).processNextWorkItem
	sigs.k8s.io/controller-runtime@v0.19.0/pkg/internal/controller/controller.go:263
sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller[...]).Start.func2.2
	sigs.k8s.io/controller-runtime@v0.19.0/pkg/internal/controller/controller.go:224
```

The difference is in a successful run, the error happens earlier than that in a fail run.
This PR tries to ignore the case to prevent the error entirely.

File: test/e2e_env/gatewayapi/conformance_test.go [hunk 1]
```diff
@@ -138,8 +139,16 @@ func TestConformance(t *testing.T) {
...
 
-	conformanceSuite.Setup(t, tests.ConformanceTests)
-	g.Expect(conformanceSuite.Run(t, tests.ConformanceTests)).To(Succeed())
+	// We are seeing flaky runs which are related to headless service cases, so ignoring them temporarily
+	// See this CI run: https://github.com/kumahq/kuma/actions/runs/10920125399/attempts/4
+	testCases := tests.ConformanceTests
+	idx := slices.IndexFunc(testCases, func(t suite.ConformanceTest) bool {
+		return t.ShortName == tests.HTTPRouteServiceTypes.ShortName
+	})
+	testCases = append(testCases[:idx], testCases[idx+1:]...)
```
Comment (michaelbeaumont) [inline test/e2e_env/gatewayapi/conformance_test.go:148]: We can just set `SkipTests` on the options

**Review by michaelbeaumont — CHANGES_REQUESTED** [review]

**Review by jakubdyszkiewicz — APPROVED** [review]

**Review by michaelbeaumont — APPROVED** [review]

**Pull request merged**